### ISSUE TITLE
Type alias causes compiler runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by advanderveer** [tinygo-org/tinygo#452]
When calling a method on the alias the compile will fail with a runtime error. To reproduce:

in file `project/a/a.go`:
```go
package a

type Attr map[string]interface{}

func (a Attr) Foo() { println("foo") }
```

in `project/main.go`
```go
package main

import (
	"project/a"
)

type At = a.Attr

func main() {
	a := At{}
	a.Foo()
}
```
Then run inside `project/`:
```
tinygo build -o wasm.wasm -target=wasm .
```

To produce:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x42e3fb1]

goroutine 1 [running]:
github.com/tinygo-org/tinygo/compiler.(*Compiler).parseCall(0xc000178000, 0xc000479e60, 0xc00036a0c0, 0x7b3f5e0, 0x0, 0x6)
	/Users/distiller/project/compiler/compiler.go:1303 +0x4a1
github.com/tinygo-org/tinygo/compiler.(*Compiler).parseExpr(0xc000178000, 0xc000479e60, 0x6d98ba0, 0xc00036a080, 0xc00036a080, 0x1, 0xb)
	/Users/distiller/project/compiler/compiler.go:1407 +0x34c5
github.com/tinygo-org/tinygo/compiler.(*Compiler).parseInstr(0xc000178000, 0xc000479e60, 0x6d98b40, 0xc00036a080)
	/Users/distiller/project/compiler/compiler.go:986 +0x167c
github.com/tinygo-org/tinygo/compiler.(*Compiler).parseFunc(0xc000178000, 0xc000479e60)
	/Users/distiller/project/compiler/compiler.go:960 +0xe2e
github.com/tinygo-org/tinygo/compiler.(*Compiler).Compile(0xc000178000, 0x7ffeefbffa4e, 0x1, 0x1b, 0x0, 0x0)
	/Users/distiller/project/compiler/compiler.go:318 +0x1007
main.Compile(0x7ffeefbffa4e, 0x1, 0x7ffeefbffa37, 0x9, 0xc00014a180, 0xc000533f00, 0xc000533d88, 0x0, 0x0)
	/Users/distiller/project/main.go:117 +0x943
main.Build(0x7ffeefbffa4e, 0x1, 0x7ffeefbffa37, 0x9, 0x7ffeefbffa49, 0x4, 0xc0000adf00, 0xc0000fe220, 0xc00010a200)
	/Users/distiller/project/main.go:323 +0xe2
main.main()
	/Users/distiller/project/main.go:616 +0xc74
```

**Comment by deadprogram** [comment]
Is this still an issue in the current `dev` branch?

**Comment by aykevl** [comment]
I can still reproduce it in the dev branch, so yes this is still an issue.
Smaller/simpler repro:

```go
package main

type Attr int

func (a Attr) Foo() {}

type At = Attr

func main() {
	var a At
	a.Foo()
}
```

**Comment by jadr2ddude** [comment]
The reason that this occurs is that the function is not discovered until the `SimpleDCE` pass, where a bug in our handling of discovery prevents it from getting marked properly. It therefore gets deleted immediately. I will hopefully get a fix PR in soon.

**Comment by jadr2ddude** [comment]
This should be fixed by #574

**Comment by jadr2ddude** [comment]
This appears to have been fixed.
